pandas/core/arrays/integer.py

```diff
@@ -371,14 +376,28 @@ def __getitem__(self, item):
             return self._data[item]
         return type(self)(self._data[item], self._mask[item])
 
-    def _coerce_to_ndarray(self):
+    def _coerce_to_ndarray(self, dtype=None, na_value=libmissing.NA):
         """
         coerce to an ndarary of object dtype
         """
-
         # TODO(jreback) make this better
-        data = self._data.astype(object)
-        data[self._mask] = self._na_value
+        if dtype is None:
+            dtype = object
+        elif is_float_dtype(dtype) and na_value is libmissing.NA:
```

Comment:
why are you exactly comparing versus NA and not just using ``isna``

Reply:
isna works on arrays as well, which would fail the if condition.

I think Joris' suggestion to use `_no_default` is good. Then we don't have to worry about NA vs. NaN vs. NaT.

Reply:
> why are you exactly comparing versus NA and not just using isna

Because `isna` catches any NA-like value (None, np.nan, pd.NaT). The goal here is that the user can specify the *exact* NA sentinel they want to use for the conversion.

Reply:
ok, great